pkg/cloud/services/gc/options.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gc
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/elb/elbiface"
+	"github.com/aws/aws-sdk-go/service/elbv2/elbv2iface"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi/resourcegroupstaggingapiiface"
+)
+
+// ServiceOption is an option for creating the service.
+type ServiceOption func(*Service)
+
+// withELBClient is an option for specifying a AWS ELB Client.
+func withELBClient(client elbiface.ELBAPI) ServiceOption {
+	return func(s *Service) {
+		s.elbClient = client
+	}
+}
+
+// withELBv2Client is an option for specifying a AWS ELBv2 Client.
+func withELBv2Client(client elbv2iface.ELBV2API) ServiceOption {
+	return func(s *Service) {
+		s.elbv2Client = client
+	}
+}
+
+// withResourceTaggingClient is an option for specifying a AWS Resource Tagging Client.
+func withResourceTaggingClient(client resourcegroupstaggingapiiface.ResourceGroupsTaggingAPIAPI) ServiceOption {
+	return func(s *Service) {
+		s.resourceTaggingClient = client
+	}
+}
+
+// withEC2Client is an option for specifying a AWS EC2 Client.
+func withEC2Client(client ec2iface.EC2API) ServiceOption {
+	return func(s *Service) {
+		s.ec2Client = client
+	}
+}
```

Comment:
This function is not called anywhere. Do we expect to call it?

The other functions are called in cleanup_test.go.

Reply:
Its called from the test file: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/f55474951bfe224ce44c4dd34e836f6edbe60c8c/pkg/cloud/services/gc/cleanup_test.go#L555

Reply:
(I'm sorry, I was reviewing the PR using an extension in vscode, and it was showing me stale data. My comment was about `WithCleanupFuncs`, which you already removed.)